### ISSUE TITLE
Fetch metadata before downloading

### ISSUE THREAD

**Issue by catzybluphish** [transmission/transmission#1017]
Is there a way to automatically download metadata when a magnet link is used but not start downloading the whole torrent. I'm requesting this feature because sometimes I don't want to download entire torrents with hundreds of episodes, so I'd like to be prompted with the files list (requiring the metadata) before starting so I can selectively choose which files to download. It's similar to qBittorrent where it opens a dialog window showing the files list before initiating the download.

**Comment by jpmhouston** [comment]
I'd like to see the current behavior for magnet links replaced by 3 new cases:

- Similar to current behavior, magnet links added are listed as a special state, metadata not fetched until later, however:
  - a new menu / contextual menu / hover button / etc. to just fetch metadata, rather than overloading the 'resume right away' operation. the menu/button should probably switch to be 'cancel fetch' when appropriate
  - otherwise, the metadata fetch then the transfer are done together, not separate steps, ie. if transfer is waiting in queue, one doesn't need to resume right away twice

- Automatically fetch metadata when added
  - determined by a setting or checkbox in the magnet link window, eg. 'fetch metadata when added' to go along with 'start when added' enabled when the latter is off
  - once magnet link added, begins start fetching metadata. can be cancelled with the 'cancel fetch' menu/button mentioned above
  - perhaps a setting for how many magnet links can be fetching their metadata at once, perhaps requiring a new internal queue of pending fetches

- Immediately fetch metadata when opened (i think this is what the OP was wanting)
  - determined by a setting, or button in the magnet link window, fetch started when that window is opened and it shows some indicator signifying that the fetch is in progress. there should be some way to cancel
  - when fetch finished, window changes into one more resembling the opened torrent file window

**Comment by livings124** [comment]
Closing in favor of #218
